crates/telio-utils/src/never.rs

```diff
@@ -0,0 +1,16 @@
+use std::{error::Error, fmt::Display};
+
+/// Error type that is impossible to construct
```

Comment:
That's not correct, it's possible to construct but it will fail at runtime when printed using `Display`.

Reply:
How would you construct it outside of this file ?

Reply:
Good point, but I think that a more obviously impossible type empty enum, something like `enum Never {}`.

Reply:
Yes I think it is more idiomatic way, but I will use Infallible, I somehow did not like at first as it comes from convert and felt like it's inteded for conversions only :sweat: but it really does not matter